### PR TITLE
Using Alpine extended image

### PR DESCRIPTION
**Pull request by AlonaKaplan**
**What this PR does / why we need it**:
The PR is using alpine extended image that contains cloud-init and ga.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by AlonaKaplan** [comment]
Summery
IPv6 single stack cluster-
1. Using **Cirros** with long timout - **~2h** (https://github.com/kubevirt/kubevirt/pull/7158)
2. Using **old Alpine** image and configuring the networking via the **console**- **~45m** (https://github.com/kubevirt/kubevirt/pull/7435)
3. Using **extended Alpine with cloud-init** - **~1h**

Dual stack cluster-
1. Using **Cirros** - **~1:25** (https://github.com/kubevirt/kubevirt/pull/7158)
2. Using **old Alpine** image and configuring the networking via the **console**- **~1:15m** (https://github.com/kubevirt/kubevirt/pull/7435)
3. Using **extended Alpine with cloud-init** - **~1:25h**

**Comment by qinqon** [comment]
> Summary IPv6 single stack cluster-
> 
> 1. Using **Cirros** with long timout - **~2h** ([Support single stack IPv6 #7158](https://github.com/kubevirt/kubevirt/pull/7158))
> 2. Using **old Alpine** image and configuring the networking via the **console**- **~45m** ([Replace Cirros with Alpine #7435](https://github.com/kubevirt/kubevirt/pull/7435))
> 3. Using **extended Alpine with cloud-init** - **~1h**
> 
> Dual stack cluster-
> 
> 1. Using **Cirros** - **~1:25** ([Support single stack IPv6 #7158](https://github.com/kubevirt/kubevirt/pull/7158))
> 2. Using **old Alpine** image and configuring the networking via the **console**- **~1:15m** ([Replace Cirros with Alpine #7435](https://github.com/kubevirt/kubevirt/pull/7435))
> 3. Using **extended Alpine with cloud-init** - **~1:25h**

Do we know where time goes at alpine console vs cloud-init ? it takes around 15m and cloud-init should run faster, maybe is the image size the issue cloud-init has a lot of dependencies since it's a python package.

File: tests/libvmi/network.go
Comment (EdDev) [inline tests/libvmi/network.go:24]: nit: Please keep a separation from the production imports.

**Review by EdDev — COMMENTED** [review]
First commit review.

> tests: Add AlpineExtended image to the tests 

Can you please elaborate and:
- Mention the reasoning/need to do so.
- Add a small note that you also added a factory for it in `libvmi` so tests can use it.

**Review by EdDev — COMMENTED** [review]
> tests, LoginToAlpine: if vmi.Spec.Hostname exists expect it in the pr…

The subject line is too long, could you please reduce it a bit?

File: tests/migration_test.go
Comment (EdDev) [inline tests/migration_test.go:2329]: Isn't it too late to do this here? It needs to be done before the `runVMIAndExpectLaunch`, no?

**Review by EdDev — COMMENTED** [review]
Final commit:

> tests: use AlpineExtended image instead of configuring Alpine via con…

Subject line is too long, could you please reduce it a bit?

---

I really like this PR, making everything so clean! :heart:

File: tests/libvmi/factory.go
Comment (AlonaKaplan) [reply tests/libvmi/factory.go:24]: done

File: tests/libvmi/network.go
Comment (AlonaKaplan) [reply tests/libvmi/network.go:24]: done

File: tests/migration_test.go
Comment (AlonaKaplan) [reply tests/migration_test.go:2329]: It is a wrong copy paste, those two lines have to be removed.
Thanks for noticing it!

**Comment by AlonaKaplan** [comment]
> First commit review.
> 
> > tests: Add AlpineExtended image to the tests
> 
> Can you please elaborate and:
> 
>     * Mention the reasoning/need to do so.
> 
>     * Add a small note that you also added a factory for it in `libvmi` so tests can use it.

done

File: tests/network/vmi_multus.go
Comment (EdDev) [inline tests/network/vmi_multus.go:311]: You sneaked in this one. Not sure it is that safe. :thinking:

**Review by EdDev — APPROVED** [review]
Thanks you!

The last two commits still have a too-long subject line (you can see it breaks when looked at the commits tab in the review).

There is also a small change that was included (see inline comment), which I am not sure if is that good to introduce here.
I will leave it to you to decide if it is safe enough.

File: tests/network/vmi_multus.go
Comment (AlonaKaplan) [reply tests/network/vmi_multus.go:311]: It is necessary since `lo` is down in the alpine image.

File: tests/containerdisk/containerdisk.go
Comment (qinqon) [inline tests/containerdisk/containerdisk.go:34]: why not following fedora's name `ContainerDiskAlpineTestTooling`

File: tests/console/login.go
Comment (qinqon) [inline tests/console/login.go:90]: This is not related

File: tests/network/expose.go
Comment (qinqon) [inline tests/network/expose.go:172]: We can start the server as part of the cloud-init with `cmd`

File: tests/network/vmi_multus.go
Comment (qinqon) [reply tests/network/vmi_multus.go:311]: Are we going to be able to replace this checking GA instead of using console ?

**Review by qinqon — CHANGES_REQUESTED** [review]
Naming nit, and some suggestiong on setting up the hello world server, also does it make sense to use GA to check interfaces are up instead of conole.

File: tests/network/vmi_multus.go
Comment (qinqon) [reply tests/network/vmi_multus.go:311]: I will try a PR

**Comment by AlonaKaplan** [comment]
/pull-kubevirt-e2e-k8s-1.23-sig-compute

File: tests/network/vmi_multus.go
Comment (phoracek) [inline tests/network/vmi_multus.go:194]: Great to see network data nicely utilized in tests

**Review by phoracek — APPROVED** [review]
/approve

**Comment by phoracek** [comment]
Since this touches BAZEL config, I can't approve the PR. The content looks fine to me.

**Comment by jean-edouard** [comment]
@AlonaKaplan Since the new image doesn't seem to have an aarch64 version, shouldn't we `SkipIfARM64()` the tests that use it?

**Comment by AlonaKaplan** [comment]
> @AlonaKaplan Since the new image doesn't seem to have an aarch64 version, shouldn't we `SkipIfARM64()` the tests that use it?

@qinqon can we create `aarch64` version of the image?

**Comment by qinqon** [comment]
> > @AlonaKaplan Since the new image doesn't seem to have an aarch64 version, shouldn't we `SkipIfARM64()` the tests that use it?
> 
> @qinqon can we create `aarch64` version of the image?

Yep, the infra was prepared to do so

**Comment by AlonaKaplan** [comment]
> @AlonaKaplan Since the new image doesn't seem to have an aarch64 version, shouldn't we `SkipIfARM64()` the tests that use it?

@jean-edouard thanks a lot for the comment! Do you know when our test are run on ARM64? Since we want to move most of our tests to use this alpine image, skipping it means that most of the tests won't run on ARM64.

**Comment by xpivarc** [comment]
> > @AlonaKaplan Since the new image doesn't seem to have an aarch64 version, shouldn't we `SkipIfARM64()` the tests that use it?
> 
> @jean-edouard thanks a lot for the comment! Do you know when our test are run on ARM64? Since we want to move most of our tests to use this alpine image, skipping it means that most of the tests won't run on ARM64.

You can see the job definition here https://github.com/kubevirt/project-infra/blob/main/github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml#L705 . It is periodic at the moment.

**Comment by jean-edouard** [comment]
> > @AlonaKaplan Since the new image doesn't seem to have an aarch64 version, shouldn't we `SkipIfARM64()` the tests that use it?
> 
> @jean-edouard thanks a lot for the comment! Do you know when our test are run on ARM64? Since we want to move most of our tests to use this alpine image, skipping it means that most of the tests won't run on ARM64.

Yeah, let's build the image for arm64 and not skip anything!

**Comment by AlonaKaplan** [comment]
> > > @AlonaKaplan Since the new image doesn't seem to have an aarch64 version, shouldn't we `SkipIfARM64()` the tests that use it?
> > 
> > 
> > @jean-edouard thanks a lot for the comment! Do you know when our test are run on ARM64? Since we want to move most of our tests to use this alpine image, skipping it means that most of the tests won't run on ARM64.
> 
> Yeah, let's build the image for arm64 and not skip anything!

@jean-edouard just realized the arm lanes are running only test with the "arm64" label. Non of the tests that are using the new "AlpineWithTestTooling" label having this label. Therefore, we can continue with this PR without introducing an "arm64" image.

Is there anything else blocking the PR?

/hold cancel

**Review by EdDev — APPROVED** [review]

**Review by qinqon — APPROVED** [review]

**Pull request merged**
